app/src/routes/Tournaments/components/TournamentProcess.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { PureComponent } from "react";
+import PropTypes from "prop-types";
+import Match from "./Match";
+import { Link } from "react-router";
+import { Button } from "react-mdl/lib";
+import TournamentFinalResultsContainer from "../containers/TournamentFinalResultsContainer";
+
+
+const propTypes = {
+    tournamentId: PropTypes.number,
+    tournamentName: PropTypes.string,
+    pairs: PropTypes.arrayOf(PropTypes.shape({
+        host: PropTypes.object.isRequired,
+        guest: PropTypes.object.isRequired,
+        winner: PropTypes.string.isRequired
+    })),
+    setWinner: PropTypes.func.isRequired,
+    nextRound: PropTypes.func.isRequired,
+    finishTournament: PropTypes.func.isRequired,
+    displayFinalResults: PropTypes.bool.isRequired,
+    toggleFinalResults: PropTypes.func.isRequired
+};
+
+const defaultProps = {
+};
+
+class TournamentProcess extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.nextRound = this.nextRound.bind(this);
+        this.finishTournament = this.finishTournament.bind(this);
+        this.toggleFinalResults = this.toggleFinalResults.bind(this);
+        this.updateNextRoundButton = this.updateNextRoundButton.bind(this);
+        this.state = {
+            tournamentFinished: false,
+            matchesFinished: this.props.pairs.filter(pair => pair.winner != "0").length
```

Comment:
keeping this in state is not necessary, you can perform this computations inside render function